src/lib.rs

```diff
@@ -1040,6 +1050,20 @@ const fn is_keyof_u16(key: &[u8]) -> bool {
     matches!(key, b"tcp" | b"tcp6" | b"udp" | b"udp6")
 }
 
+fn is_valid_ipv4(key: &[u8], value: &[u8]) -> bool {
+    if key == b"ip" && value[0] - 0x80 == 0x04 {
```

Comment:
I see a couple issues with this lines:
- `value[0]` will panic if the slice is empty
- `value[0] - 0x80` will overflow (an panic on debug) in a good portion of the range in u8
- `value[0] - 0x80 == 0x04` doesn't seem to guarantee this is a valid ipv4